### ISSUE TITLE
Using the proxy

### ISSUE THREAD

**Issue by akillar** [warren-bank/HLS-Proxy#8]
How do you use this proxy to streama m3u8 url. i've installed the proxy globally and started the proxy using the command + allwed ports in the firewall. but how do i use my m3u8 url to use this proxy ?

**Comment by warren-bank** [comment]
using the 24/7 CBS News live stream as a sample [m3u8 URL](https://www.cbsnews.com/common/video/cbsn_header_prod.m3u8)

to obtain a proxied URL in bash:
```bash
proxy='127.0.0.1:8080'
URL='https://www.cbsnews.com/common/video/cbsn_header_prod.m3u8'
URL=$(echo "$URL" | base64 --wrap=0)
URL="http://${proxy}/${URL}.m3u8"
echo $URL
```

to obtain a proxied URL in javascript:
```javascript
{
  const proxy = '127.0.0.1:8080'
  let URL
  URL = 'https://www.cbsnews.com/common/video/cbsn_header_prod.m3u8'
  URL = window.btoa(URL)
  URL = `http://${proxy}/${URL}.m3u8`
  console.log(URL)
}
```

tool that can be used to simplify this task:
* [Chrome extension](https://github.com/warren-bank/crx-webcast-reloaded) that can run in any Chromium-based desktop web browser, and be used to:
  - intercept the URL of (nearly) all videos on any website
  - display these video URLs as a list of links
    * clicking on [this icon](https://github.com/warren-bank/crx-webcast-reloaded/raw/v0.6.0/chrome_extension/data/proxy.png) will transfer the URL of the video to [this single-page application (SPA)](http://webcast-reloaded.surge.sh/proxy.html) that can run in any web browser
      - it allows you to specify the location for your running instance of HLS Proxy
      - when you click "Load Player", it will:
        * construct a proxied URL
        * redirect to [another SPA](http://webcast-reloaded.surge.sh/index.html)
          - this page lets you choose where to redirect this proxied URL:
            * Chromecast sender
              - also includes an in-browser video player to watch the proxied HLS stream
            * AirPlay sender
            * HLS Proxy configuration
              - this link would be redundant, since you've already configured an HLS Proxy

a cheat to use the SPAs without the Chrome extension:
* open [this page](http://webcast-reloaded.surge.sh/1-webcast/index.html)
  - enter the URL for your HLS manifest
    * example: `https://www.cbsnews.com/common/video/cbsn_header_prod.m3u8`
  - click "Watch" and a hash is added to the URL in the browser's address bar
    * example: `http://webcast-reloaded.surge.sh/1-webcast/index.html#/watch/aHR0cHM6Ly93d3cuY2JzbmV3cy5jb20vY29tbW9uL3ZpZGVvL2Nic25faGVhZGVyX3Byb2QubTN1OA%253D%253D`
  - manually modify this URL to replace `1-webcast/index.html` with `proxy.html`
    * example: `http://webcast-reloaded.surge.sh/proxy.html#/watch/aHR0cHM6Ly93d3cuY2JzbmV3cy5jb20vY29tbW9uL3ZpZGVvL2Nic25faGVhZGVyX3Byb2QubTN1OA%253D%253D`
  - follow the same instructions for using the Chrome extension, beginning at the step immediately after having clicked on [this icon](https://github.com/warren-bank/crx-webcast-reloaded/raw/v0.6.0/chrome_extension/data/proxy.png)

**Comment by warren-bank** [comment]
PS: since you mentioned opening ports in your server's firewall, I'll mention that you'll want to replace `127.0.0.1` in all of the examples above with an IP that's reachable from your client. For example, on your LAN the IP might look something like `192.168.0.100`.

**Comment by akillar** [comment]
thank you @warren-bank.

**Comment by akillar** [comment]
@warren-bank is it possible to use a domain name instead of a ip address for the  **const proxy = '127.0.0.1:8080'**

**Comment by warren-bank** [comment]
yes.. so long as the client that you're using to download the stream through the proxy can use DNS to resolve your domain name to an IP address, then you certainly can.

```javascript
{
  const proxy = 'https://my-hls-proxy.com:443'
  let URL
  URL = 'https://www.cbsnews.com/common/video/cbsn_header_prod.m3u8'
  URL = window.btoa(URL)
  URL = `${proxy}/${URL}.m3u8`
  console.log(URL)
}
```
